docs/source/developers/guide/step_by_step/arrow_codebase.rst

```diff
@@ -0,0 +1,152 @@
+connected to that language. For example:
+
+- The ``python/`` folder includes ``pyarrow/`` folder which contains
+  the code for the pyarrow package and requirements files that you
+  need when building pyarrow.
+
+  The ``pyarrow/`` includes Python and Cython code.
+
+  The ``pyarrow/`` also includes ``test/`` folder where all the tests
+  for the pyarrow modules are located.
+
+- The ``R/`` folder holds the code and the documentation for the R package.
+
+  The documentation can be found in ``vignettes/`` and in ``R/`` folder
+  you can find the reference documentation.
+
+  The ``R/`` folder also includes tests for the R package
+  in ``tests/`` and ``extra-tests/``.
+
+Other subdirectories included in the arrow repository are:
+
+- ``ci/`` contains code connected to the continuous integration work.
+- ``dev/`` contains scripts useful to developers when packaging,
+  testing, or committing to Arrow.
+- ``docs/`` contains most of the documentation. Read more on
+  :ref:`documentation`.
+- ``format/`` contains the Arrow Protocol files.
+
+Other files included in Arrow are connected to either GitHub,
+CI builds or Docker.
+
+Bindings, features, fixes and tests
+===================================
+
+You can read through this section to get some ideas on how
+to work around the library on the issue you have.
+
+Depending on the problem you want to solve (adding a simple
+binding, adding a feature, writing a test, …) there are
+different ways to get the necessary information. 
+
+**For all the cases** you can help yourself with
+searching for functions via some kind of search tool.
+In our experience there are two good ways:
+
+#. Via **GitHub Search** in the Arrow repository (not a forked one)
+   This way is great as GitHub lets you search for function
+   definitions and references also.
+
+#. **IDE** of your choice.
+
+**Binding**
+
+Binding means that the function in the C++ implementation is connected from 
+other languages (C (Glib), MATLAB, Python, R or Ruby). Once a 
+function is defined in the C++ we connect it from other languages
+so that it can be used there also.
+
+.. note::
+	There is much you can learn with checking **Pull Requests**
+	and **unit tests for similar issues**.  
+
+.. tabs::
+
+   .. tab:: Python
+
+      **Adding a fix in Python**
+
+      If you are doing a correction of an existing function, the
+      easiest way is to run Python interactively or run Jupyter
+      Notebook and research
+      the issue until you understand what needs to be done.
+
+      After, you can search on GitHub for the function name, to
+      see where the function is defined.
+
+      Also, if there are errors produced, the errors will most
+      likely point you towards the file you need to take a look at.
+
+      **Python - Cython - C++**
+       
+      It is quite likely that you will bump into Cython code when
+      working on Python issues. Less likely is that C++ code would
+      need some correction, but it can happen.
```

Comment:
```suggestion
      It is quite likely that you will bump into Cython code when
      working on Python issues. It's less likely is that the C++ code 
      needs updating, though it can happen.
```